specification/analysisservices/resource-manager/Microsoft.AnalysisServices/2017-08-01-beta/analysisservices.json

```diff
@@ -381,6 +381,48 @@
       }
     }
   },
+  "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.AnalysisServices/servers/{serverName}/listGatewayStatus": {
+      "post": {
+        "description": "Return the gateway status of the specified Analysis Services server instance.",
+        "tags": [ "Servers" ],
+        "operationId": "Servers_ListGatewayStatus",
+        "parameters": [
+          {
+            "$ref": "#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "name": "serverName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "pattern": "^[a-z][a-z0-9]*$",
+            "minLength": 3,
+            "maxLength": 63,
+            "description": "The name of the Analysis Services server. It must be at least 3 characters in length, and no more than 63."
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK.",
+            "schema": { "$ref": "#/definitions/GatewayListStatusLive" }
+          },
+          "400": {
```

Comment:
If 4xx responses are supposed to throw an exception in generated code (it looks like that was the intention), they can unfortunately not be modeled like that but have to go into a `default` response. See https://github.com/Azure/azure-rest-api-specs/blob/current/documentation/creating-swagger.md#negative-responses for details. I'd recommend documenting status codes and their corresponding description in the description of the default response.